src/server/controllers/user.js

```diff
@@ -34,16 +37,17 @@ const postLoginDetails = async (req, res) => {
                 username
             }
         });
-        if (findUser) {
-            if (await bcrypt.compare(password, findUser.password)) {
-                const token = jwt.sign(username, secret);
-                return res.json({ userToken: token })
-            } else {
-                return res.json("Username or Password is incorrect.")
-            }
-        } else {
-            return res.json("Username or Password is incorrect.");
+
+        const matchingPassword = await bcrypt.compare(password, findUser.password);
+
+        if (findUser && matchingPassword) {
+            const token = jwt.sign(username, secret);
+            console.log(token)
+            return res.json({ data: token })
```

Comment:
This works so much better. Nice work. We should put half a day aside next week to test all the possible errors the user can make and make sure were handling them properly.

Reply:
Agreed, currently there is a json error when defining matchingPassword if findUser is undefined, so need to make defining matchingPassword conditional.

Reply:
Sounds like a plan!